docs/developer-guide/frontend-testing.md

```diff
@@ -0,0 +1,184 @@
+
+We use different Playwright configurations for `app` and `component` testing.
+
+- `ui/test/app.config.ts`: App test configuration
+- `ui/test/component.config.ts`: Component test configuration
+
+We do this because we modify the base configuration that comes with Playwright so that component testing works, however this configuration is incompatible with app testing thus we use 2 configurations.
+
+#### Shared test package
+
+Both the `app` and `component` tests depend on the `@openremote/test` package which includes shared fixtures, configurations and our Playwright component testing plugin.
+
+The `shared` fixture in the test package is meant for general test utilities like intercepting requests.
+
+Each project that needs testing should configure its own Playwright configuration file which must reuse the above-mentioned configurations.
+
+The plugin for component testing mimics Playwright’s component testing plugin, which normally comes with Vite, but this is incompatible with the `commonjs` imports used in some components. Playwright uses Vite to bundle and mount a component to an empty HTML document for testing. Our Playwright plugin mimics the Vite based plugin using Rspack so we can mount our components to the document without import issues.
+
+### Component test setup
+
+The component tests are used to test individual Lit web components.
+
+#### Configuration
+
+Each component runs its own tests in parallel.
+
+- **Target:** Any component in the `ui/component/*` directory.
+- **Runner App:** The component test setup includes a dedicated app at `ui/test/playwright` (used to display components and serving static files).
+- **Testing Strategy:** Based on Playwright’s experimental component testing API, but adapted to Rspack using a custom plugin. Each component may include a `fixtures` directory which can provide reusable test code for other components and apps.
+
+### End-to-End test setup
+
+The end-to-end tests are used to test app functionality from the frontend.
+
+#### Configuration
+
+The end-to-end tests run sequentially using 1 worker to avoid tests interfering with each other.
+
+- **Target:** Any app in the `ui/app` directory.
+- **Worker Scope:** Single worker (to avoid tests interfering with one-another).
+- **Fixtures:** The Manager app includes a `fixtures` directory with test and data fixtures.
+- **Setup & Teardown:** The e2e test projects depend on the `*.setup.ts` and `*.cleanup.ts` project files. These projects should be used to provision realm(s), user(s) and collect authentication states for more robust and performant tests.
+
+```ts
+function createAppSetupAndTeardown(app) {
+  return [
+    {
+      name: `setup ${app}`,
+      testMatch: "**/*.setup.ts",
+      teardown: `cleanup ${app}`,
+      worker: 1,
+    },
+    {
+      name: `cleanup ${app}`,
+      testMatch: "**/*.cleanup.ts",
+      worker: 1,
+    },
+  ];
+}
+```
+
+## Writing tests
+
+### Prerequisites
```

Comment:
I would split this up in "**Prerequisites**", and "**Creating a test file**", or something like that.